consolidate.js

```diff
@@ -0,0 +1,36 @@
+var utils = require('./utils')
+
+// similar to accumulative but add all inputs
+module.exports = function consolidate (utxos, outputs, feeRate) {
+  if (!isFinite(utils.uintOrNaN(feeRate))) return {}
+  var bytesAccum = utils.transactionBytes([], outputs)
+
+  var inAccum = 0
+  var inputs = []
+  var outAccum = utils.sumOrNaN(outputs)
+
+  for (var i = 0; i < utxos.length; ++i) {
+    var utxo = utxos[i]
+    var utxoBytes = utils.inputBytes(utxo)
+    var utxoFee = feeRate * utxoBytes
+    var utxoValue = utils.uintOrNaN(utxo.value)
+
+    // skip detrimental input
+    if (utxoFee > utxo.value) {
```

Comment:
If I have a 1000 sat utxo and it adds a 1000 sat fee to my transaction, this will add it...

Is this assumption also made in other parts of the library? If so that's fine, but if not, I would perhaps add a parameter for the user to define a percentage of fee that is the cutoff.

IMO if we have a utxo of 1000 value, the highest acceptable fee should be 999 and the lowest should be 0. The user of this API should be able to decide whether they want to skip above 10 or skip above 900.

Reply:
This is copypasted from https://github.com/bitcoinjs/coinselect/blob/becfd26096fd9fc719f5b4928935be61d87bbb98/accumulative.js#L20 I did not assume anything